### ISSUE TITLE
Add support for function(require, exports, module){} syntax

### ISSUE THREAD

**Issue by kirill-konshin** [scalableminds/amd-optimize#34]
Currently code like

```js
define(function(require, exports, module) {

    require('angular-ui-router');

    module.exports = require('angular').module('ui.router');

});
```

will be transformed into

```js
define('contrib/angular-ui-router', [
    'angular-ui-router',
    'angular'
], function (require, exports, module) {
    require('angular-ui-router');
    module.exports = require('angular').module('ui.router');
});
```

Which obviously will cause errors.

I might be wrong, but it is supposed to be something like:

```js
define('contrib/angular-ui-router', [
    'require', 'exports', 'module',
    'angular-ui-router',
    'angular'
], function (require, exports, module) {
    require('angular-ui-router');
    module.exports = require('angular').module('ui.router');
});
```

**Comment by kirill-konshin** [comment]
Plus my recent `text` plugin patch works weird with this kind of notation:

```js
define(function(require, exports, module){
   var html = require('text!./file.html'); // this is not touched although there is a definition for 'file.html', basically, text! part should be simply removed
```

**Comment by komputerwiz** [comment]
Same issue here, too. I can verify that @kirill-konshin's solution (prepending `'require', 'exports', 'module'` to the deps list) works: see [source code for Almond (as of 37b54ea) line 253 and below](https://github.com/jrburke/almond/blob/37b54ea1463b9583fa0370a323907a7cefbd738f/almond.js#L253).

**Comment by normanrz** [comment]
This was fixed by @komputerwiz's PR #37. Thanks!
